.circleci/config.yml

```diff
@@ -1,73 +1,121 @@
+references:
+  #
+  # Environment
+  #
+  shared-environment: &shared-environment
+    CIRCLE_ARTIFACTS: /tmp/artifacts
+    CIRCLE_TEST_REPORTS: /tmp/test_results
+
+  #
+  # Image
+  #
+  docker-image: &docker-image
+    - image: circleci/node:10.10.0-browsers
+
+  #
+  # Artifact and test results
+  #
+  setup-results-and-artifacts: &setup-results-and-artifacts
+    name: Create Directories for Results and Artifacts
+    command: |
+      mkdir -p                             \
+        "$CIRCLE_ARTIFACTS/translate"      \
+        "$CIRCLE_TEST_REPORTS/client"      \
+        "$CIRCLE_TEST_REPORTS/integration" \
+        "$CIRCLE_TEST_REPORTS/server"
+
+
+  #
+  # Cache management
+  #
+  restore-git-cache: &restore-git-cache
+    name: Restore git cache
+    key: v1-20180524-git
+  save-git-cache: &save-git-cache
+    name: Save git cache
+    key: v1-20180524-git
+    paths:
+      - ".git"
+
+  restore-npm-cache: &restore-npm-cache
+    name: "Restore npm cache"
+    keys:
+       - v1-npmcache-{{ checksum ".nvmrc" }}-{{ checksum "npm-shrinkwrap.json" }}
+       - v1-npmcache-{{ checksum ".nvmrc" }}
+  save-npm-cache: &save-npm-cache
+    name: "Save npm cache"
+    key: v1-npmcache-{{ checksum ".nvmrc" }}-{{ checksum "npm-shrinkwrap.json" }}
+    paths:
+      - ~/.npm
+
+  defaults: &defaults
+    working_directory: ~/wp-calypso
+    docker: *docker-image
+    environment: *shared-environment
+
 version: 2
 jobs:
   build:
+    <<: *defaults
     parallelism: 6
-    working_directory: ~/wp-calypso
-    docker:
-      - image: circleci/node:10.10.0-browsers
-        environment:
-          CIRCLE_ARTIFACTS: /tmp/artifacts
-          CIRCLE_TEST_REPORTS: /tmp/test_results
 
     steps:
-      - restore_cache:
-          name: Restore git cache
-          key: v1-20180524-git
-
+      ###############
+      # Basic Setup #
```

Comment:
I'd love for this section to be shared, but YAML doesn't appear to allow merging arrays, so a sequence of steps cannot be merged like:

```yaml
steps:
  <<: *base-steps
  - another-step
```

Reply:
Hmm, really? The [docs](https://circleci.com/docs/2.0/writing-yaml/#merging-maps) seem to suggest that's possible, or am I misreading them?

Reply:
🤔 hmmm…

So, we can't do my example, but maybe we can do like it suggests in that do. I understand we'd need to make another reference for the "main" steps of each job. I'll see if that works…

Reply:
278b634 and f5992a5

This just doesn't seem to be possible.